src/module/skb/bpf/skb_hook.bpf.c

```diff
@@ -371,6 +382,85 @@ static __always_inline int process_skb_l2(struct retis_raw_event *event,
 	return 0;
 }
 
+static __always_inline int process_packet(struct retis_raw_event *event,
+					  struct sk_buff *skb)
+{
+	unsigned char *head, *data;
+	struct skb_packet_event *e;
+	int size, data_offset;
+	u32 len, linear_len;
+	u16 mac, network;
+
+	data = BPF_CORE_READ(skb, data);
+	head = BPF_CORE_READ(skb, head);
+
+	data_offset = data - head;
+	if (data_offset < 0) /* Keep the verifier happy */
+		return 0;
+
+	mac = BPF_CORE_READ(skb, mac_header);
+	network = BPF_CORE_READ(skb, network_header);
+	len = BPF_CORE_READ(skb, len);
+	linear_len = len - BPF_CORE_READ(skb, data_len); /* Linear buffer size */
+
+	/* Best case: mac offset is set and valid */
+	if (mac != (u16)~0U && (!network || (network && mac != network))) {
```

Comment:
other places requiring valid and existing mac are simply considering `(mac != (u16)~0U)`, should we also extend (maybe the existing helper) the check for those as well?